src/jdk.incubator.vector/share/classes/jdk/incubator/vector/X-Vector.java.template

```diff
@@ -823,7 +817,16 @@ public abstract class $abstractvectortype$ extends AbstractVector<$Boxtype$> {
     $abstractvectortype$ lanewise(VectorOperators.Binary op,
                                   $type$ e,
                                   VectorMask<$Boxtype$> m) {
-        return blend(lanewise(op, e), m);
+#if[BITWISE]
+        if (opKind(op, VO_SHIFT) && ($type$)(int)e == e) {
+            $abstractvectortype$ shift = lanewiseShift(op, (int) e);
+            return m != null ? blend(shift, m) : shift;
```

Comment:
Checking for m!=null would cause boxing here.

Reply:
Since I didn't add the masked intrinisic version for `broadcastInt` that is called by `lanewiseShift`. I think it can be fixed with the same solution for other binary `lanewise`. Once the masked `broadcastInt` is added, the code might look like:
```
   if (opKind(op, VO_SHIFT) && ($type$)(int)e == e) {
         return lanewiseShift(op, (int) e, m);
   }
```

Reply:
sounds good.